botv2.py

```diff
@@ -0,0 +1,160 @@
+import discord
+import os
+import json
+
+from discord.ext.commands import Bot
+from checks import is_year
+from discord.ext import commands
+
+colours = []
+roles = []
+guild = None
+desc = '''A bot made by server admins to manage KeyGEnCoders discussion Server'''
+bot = Bot(command_prefix=commands.when_mentioned_or("!"),description=desc)
+
+
+def get_colours():
+    f = open("colour.json",'r')
+    c = json.loads(f.read())
+    for i in range(1, 6):
+        colours.append(discord.Colour(int(c[str(i)], 16)))
+
+
+def get_year(member):
+    roles = member.roles
+    for role in roles:
+        if is_year(role.name):
+            return role.name
+
+
+get_colours()
+
+
+@bot.event
+async def on_message(message):
+    await bot.process_commands(message)
+
+
+@bot.event
+async def on_ready():
+
+    global roles
+    global guild
+    guild = bot.guilds
+
+    guild = guild[0]
```

Comment:
Don't assume the bot will be in one guild only. After adding to the main guild there will two guilds the bot is part of.

Reply:
will kick the bot from test? or else have to make a lot of changes now.